crates/core/src/agent/state.rs

```diff
@@ -270,15 +287,39 @@ fn reduce_commit_entry(
         Ok(address)
     });
 
+    agent_state.actions.insert(
+        action_wrapper.clone(),
+        Response::from(AgentActionResponse::Commit(result)),
+    );
+}
+
+fn reduce_prune(agent_state: &mut AgentState, _root_state: &State, action_wrapper: &ActionWrapper) {
+    assert_eq!(action_wrapper.action(), &Action::Prune);
```

Comment:
Since `action_wrapper` is only used here in this function, would it make sense to pass `prune: &Action::Prune` directly, instead of `action_wrapper: &ActionWrapper`?

Reply:
We have `ActionWrapper` as the type that gets processed by the action loop. As stated above, the difference is a `ProcessUniqueId` to make actions unique. We can't (easily) change that for just one reducer.